src/libs/actions/Report.js

```diff
@@ -168,6 +168,53 @@ function getChatReportName(fullReport, chatType) {
         .join(', ');
 }
 
+/**
+ * Returns the number of optimisticReportActionIDs in reportActions
+ *
+ * @param {Number} reportID
+ * @returns {Number}
+ */
+function getNumberOfOptimisticActions(reportID) {
+    const optimisticActionIds = lodashGet(allReports, [reportID, 'optimisticReportActionIDs'], []);
+    return _.filter(optimisticActionIds, actionId => ReportActions.doesOptimisticReportActionExist(reportID, actionId)).length;
+}
+
+/**
+ * If onyx local report data is newer than fetched report then return updated version of simplifiedReport
+ * Local report data could be newer than the fetched report because of a pusher new comment event coming while the fetching report request hasn't been completed yet
+ *
+ * @param {Object} simplifiedReport, simplifiedReport of Report Data
+ * @returns {Object}
+ */
+function checkAndUpdateSimplifiedReport(simplifiedReport) {
+    const currentReport = lodashGet(allReports, simplifiedReport.reportID, 0);
+    if (!currentReport) {
+        return simplifiedReport;
+    }
+    const simplifiedReportMaxSeqNumber = lodashGet(simplifiedReport, 'maxSequenceNumber', 0);
+    const currentReportMaxSeqNumber = lodashGet(currentReport, 'maxSequenceNumber', 0);
+
+    // Exclude optimisticActionIds from maxSequenceNumber of current local report when comparing maxSequenceNumber of fetched report and local onyx report.
+    // Optimistic report actions is a new comment not yet sent to the server.
+    const currentMaxSeqNumberWithoutOptimisticActions = currentReportMaxSeqNumber - getNumberOfOptimisticActions(currentReport.reportID);
+    if (simplifiedReportMaxSeqNumber >= currentMaxSeqNumberWithoutOptimisticActions) {
+        return simplifiedReport;
+    }
+
+    const lastReadSequenceNumber = lastReadSequenceNumbers[simplifiedReport.reportID];
+    const unreadActionCount = currentMaxSeqNumberWithoutOptimisticActions - lastReadSequenceNumber - ReportActions.getDeletedCommentsCount(simplifiedReport.reportID, lastReadSequenceNumber);
+
+    // Update some fields that are set by updateReportWithNewAction, the handler of pusher new comment event.
+    const updatedSimplifiedReport = _.clone(simplifiedReport);
+    updatedSimplifiedReport.maxSequenceNumber = currentReport.MaxSeqNumber;
```

Comment:
`currentReport.MaxSeqNumber` I am curious to know where are we setting this property?